pandas/tests/io/test_s3.py

```diff
@@ -27,3 +28,8 @@ def test_streaming_s3_objects():
     for el in data:
         body = StreamingBody(BytesIO(el), content_length=len(el))
         read_csv(body)
+
+
+def test_parse_s3_url_with_pound_sign():
```

Comment:
this is not a useful test, you need to mock the routines as above

Reply:
Thanks for the review.  I have to disagree with you about this not being a useful test.  It tests the buggy function, fails without the fix and passes with the fix.  Can you provide any more hints about how I would mock this as you suggested?  I can't see how to do it.

Reply:
look at any tests in pandas/tests/io//parser/test_network.py, you need to use the s3_resource fixture.

Reply:
Thanks, I believe I figured it out now.  :)  `moto` seems very nice, I hadn't heard of it before.